### PR TITLE
Fix visibility checking

### PR DESCRIPTION
**Pull request by gacholio**
- handle exceptions and pop frames correctly after visibility checks
- update new instance prototype access check
- fix confusion between resolve and lookup flags
- add "no java" lookup flag to ensure no java code runs

Fixes: #2948

[ci skip]

Signed-off-by: Graham Chapman <graham_chapman@ca.ibm.com>

### CODE REVIEW

File: runtime/codert_vm/cnathelp.cpp [hunk 1]
```diff
@@ -2448,50 +2448,59 @@ old_slow_jitNewInstanceImplAccessCheck(J9VMThread *currentThread)
...
+	/* See if the method is visible - don't use checkVisibility here because newInstance
+	 * has different rules for protected method visibility.
+	 */
+	if (!J9_ARE_ANY_BITS_SET(romMethod->modifiers, J9AccPublic)) {
+		if (J9_ARE_ANY_BITS_SET(romMethod->modifiers, J9AccPrivate)) {
+			/* Private methods are visible only within their declaring class
+			 * unless the two classes are in the same nest (JDK11 and beyond).
+			 */
+#if defined(J9VM_OPT_VALHALLA_NESTMATES)
+			if (thisClass->nestHost != callerClass->nestHost)
```
Comment (DanHeidinga) [inline runtime/codert_vm/cnathelp.cpp]: This is insufficient for the nest mates case as the `nestHost` may not have been loaded and validated yet.  It likely needs to call `loadAndVerifyNestHost()` if the nestHosts are null.

File: runtime/vm/resolvesupport.cpp [hunk 2]
```diff
@@ -1876,6 +1895,7 @@ resolveMethodHandleRefInto(J9VMThread *vmThread, J9ConstantPool *ramCP, UDATA cp
 	cpItem = (J9ROMMethodRef *) &(J9_ROM_CP_FROM_CP(ramCP)[fieldOrMethodIndex]);
 	definingClass = ((J9RAMClassRef *) &(ramCP[cpItem->classRefCPIndex]))->value;
 	if (definingClass == NULL) {
+		/* TODO: no exception set here */
```
Comment (DanHeidinga) [inline runtime/vm/resolvesupport.cpp]: TODO?

File: runtime/codert_vm/cnathelp.cpp (hunk 1 above)
Comment (gacholio) [reply runtime/codert_vm/cnathelp.cpp]: At the start of the function, checkVisibility is called with thisClass and callerClass, which should ensure the loading, shouldn't it?

File: runtime/vm/resolvesupport.cpp (hunk 2 above)
Comment (gacholio) [reply runtime/vm/resolvesupport.cpp]: Throwing NPE now, if requested.

File: runtime/codert_vm/cnathelp.cpp (hunk 1 above)
Comment (DanHeidinga) [reply runtime/codert_vm/cnathelp.cpp]: Classes are either `public` or `package` access at the VM level.  Never `private` which is the only case that can cause nest validation to occur.

**Comment by DanHeidinga** [comment]
Once the nestmates check is ironed out, the rest of this looks fine

File: runtime/codert_vm/cnathelp.cpp (hunk 1 above)
Comment (gacholio) [reply runtime/codert_vm/cnathelp.cpp]: Should be fixed now.

**Comment by DanHeidinga** [comment]
Jenkins test sanity plinux jdk11

**Comment by DanHeidinga** [comment]
Jenkins test extended xlinux jdk8

**Comment by pshipton** [comment]
@gacholio Note the compile failure.

**Comment by gacholio** [comment]
Jenkins test sanity plinux jdk11

**Comment by DanHeidinga** [comment]
Jenkins test extended xlinux jdk8

**Comment by gacholio** [comment]
Jenkins test sanity plinux jdk11

**Comment by gacholio** [comment]
Jenkins test extended xlinux jdk8

**Pull request merged**